migration/idempotent/008-tracing-functions.sql

```diff
@@ -569,6 +569,39 @@ $func$
 LANGUAGE plpgsql;
 GRANT EXECUTE ON FUNCTION ps_trace.put_instrumentation_lib(text, text, bigint) TO prom_writer;
 
+-- Creates a temporary table (if it doesn't exist), suitable for tracing data ingestion.
+-- Supresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
+CREATE OR REPLACE FUNCTION _ps_trace.ensure_trace_ingest_temp_table(_temp_table_name text, _proto_table_name text)
+    RETURNS void
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+DECLARE
+    _log_setting text;
+BEGIN
+    IF _proto_table_name NOT IN ('span', 'link', 'event') THEN
+        RAISE EXCEPTION 'Invalid table prototype specified: %', _proto_table_name;
+    END IF;
+
+    _log_setting = current_setting('log_statement');
+    SET log_statement = 'none';
+
+    EXECUTE format($sql$CREATE TEMPORARY TABLE IF NOT EXISTS %I ON COMMIT DELETE ROWS AS TABLE _ps_trace.%I WITH NO DATA$sql$,
```

Comment:
Any reason we actually avoid using `CREATE TABLE x(LIKE y);`?

Reply:
I've copied @jgpruitt code. Is there a difference?

Reply:
That code was actually there already. I just modified it from `ON COMMIT DROP` to `ON COMMIT DELETE ROWS`.

This is interesting. This code is using the `CREATE TABLE AS` [statement](https://www.postgresql.org/docs/current/sql-createtableas.html) combined with the `TABLE` [statement](https://www.postgresql.org/docs/current/sql-select.html#SQL-TABLE). `TABLE foo` == `select * from foo`.

The `CREATE TABLE LIKE` statement would do essentially the same thing without running a `select * from foo`. Although, it looks like the LIKE route also copies not null constraints and the CREATE TABLE AS doesn't appear to.

Reply:
I think we should change this to use the LIKE approach. Running a needless `SELECT *` seems like a bad idea.